### PR TITLE
[TCling] Do not add decls for control statements if already annotated 

### PR DESCRIPTION
**Pull request by devajithvs**
# This Pull request:

## Checklist:

- [ ] tested changes locally
- [ ] updated the docs (if necessary)

This PR fixes https://github.com/root-project/root/issues/8367

### CODE REVIEW

**Comment by devajithvs** [comment]
Not a proper solution.

What we ideally want to do is check if a `VarDecl` with the same identifier is already added for the current `ControlScope`.

**Review by vgvassilev — COMMENTED** [review]
Any chance in adding a test?

**Comment by dpiparo** [comment]
I also think a test would be good. @vgvassilev in which subdirectory would you put the simple macro which was crashing in https://github.com/root-project/roottest/tree/master/cling ?

**Comment by vgvassilev** [comment]
Maybe we can do a death test in the test folder of core/metacling.

**Comment by dpiparo** [comment]
rebased & test added

**Review by dpiparo — APPROVED** [review]
LGTM: thanks a lot for this fix. Once the tests are green, afaik we are ready to merge.

**Comment by dpiparo** [comment]
Can it be that this fix fixes also https://github.com/root-project/root/issues/15537 ? That's what I see, I think...

**Comment by devajithvs** [comment]
As a note, once we remove `auto auto` support in: https://github.com/root-project/root/pull/16410, this functionality will no longer be needed and will need to be reverted/rebased.

**Review by vgvassilev — APPROVED** [review]
Lgtm!

**Comment by hahnjo** [comment]
> Can it be that this fix fixes also #15537 ? That's what I see, I think...

Still fails for me. It seems important to put `badcast` into a macro...

File: core/metacling/src/TClingCallbacks.cxx [hunk 1]
```diff
@@ -845,6 +845,13 @@ bool TClingCallbacks::tryResolveAtRuntimeInternal(LookupResult &R, Scope *S) {
       return false;
    }
 
+   // FIX-8367
+   // Do not try to resolve if/switch/while/for statement, if already annotated
+   // for(i=0;i<0;); was failing due to multiple addDecls for the same variable.
```
Comment (hahnjo) [inline core/metacling/src/TClingCallbacks.cxx:850]: I believe we should not litter the source code with references to bug reports. If there's an explanation for this code that works without, can you maybe edit this comment?

**Review by hahnjo — APPROVED** [review]
Thanks!

**Pull request merged**